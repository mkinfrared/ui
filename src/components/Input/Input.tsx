import { InputProps } from "./Input.type";

const Input = ({ className, inputRef, value, ...rest }: InputProps) => {
  const testid = "Input";

  if (typeof value !== "undefined") {
    return (
      <input
        {...rest}
        className={className}
        data-testid={testid}
        ref={inputRef}
        value={value}
      />
    );
  }

  return (
    <input
      {...rest}
      className={className}
      data-testid={testid}
      ref={inputRef}
    />
  );
};

export default Input;
