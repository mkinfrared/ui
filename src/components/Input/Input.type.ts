import { InputHTMLAttributes, Ref } from "react";

type InputAttributes = InputHTMLAttributes<HTMLInputElement>;

export type InputProps = Omit<InputAttributes, "type"> & {
  inputRef?: Ref<HTMLInputElement>;
  type?:
    | "email"
    | "file"
    | "hidden"
    | "number"
    | "password"
    | "search"
    | "tel"
    | "text"
    | "url";
};
