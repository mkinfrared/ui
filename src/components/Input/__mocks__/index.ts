export { default } from "./Input.mock";
