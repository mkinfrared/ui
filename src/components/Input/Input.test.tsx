import { render } from "@testing-library/react";

import Input from "./Input";

describe("<Input />", () => {
  const Component = <Input />;

  it("should be defined", () => {
    expect(Input).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Input");

    expect(element).toBeDefined();
  });

  it("should set value on input if it's not undefined", () => {
    const { getByTestId } = render(<Input value="marklar" />);
    const element = getByTestId("Input");

    expect(element).toHaveValue("marklar");
  });
});
