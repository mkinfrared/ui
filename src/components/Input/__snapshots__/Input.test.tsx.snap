// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match the snapshot 1`] = `
<div>
  <input
    data-testid="Input"
  />
</div>
`;
